contracts/PercentageTransferManager/lib.rs

```diff
@@ -0,0 +1,212 @@
+    }
+
+    #[ink(event)]
+    struct ChangePrimaryIssuance {
+        #[ink(topic)]
+        allow_primary_issuance: bool
+    }
+
+    // #[ink(event)]
+    // struct ModifyExemptionList {
+    //     #[ink(topic)]
+    //     identity: IdentityId,
+    //     #[ink(topic)]
+    //     exempted: bool
+    // }
+
+    #[ink(event)]
+    struct TransferOwnership {
+        #[ink(topic)]
+        new_owner: AccountId,
+        #[ink(topic)]
+        old_owner: AccountId
+    }
+
+    /// Copy of the custom type defined in `/runtime/src/template.rs`.
+    ///
+    /// # Requirements
+    /// In order to decode a value of that type from the runtime storage:
+    ///   - The type must match exactly the custom type defined in the runtime
+    ///   - It must implement `Decode`, usually by deriving it as below
+    ///   - It should implement `Metadata` for use with `generate-metadata` (required for the UI).
+    #[derive(scale::Decode, scale::Encode)]
+    #[cfg_attr(feature = "ink-generate-abi", derive(type_metadata::Metadata))]
+    pub enum RestrictionResult {
+        Valid,
+        Invalid,
+        ForceValid
+    }
+
+    impl PercentageTransferManagerStorage {
+        /// Constructor that initializes the `u128` value to the given `max_allowed_percentage`
+        /// & boolean value for the `allow_primary_issuance`.
+        #[ink(constructor)]
+        fn new(&mut self, max_percentage: u128, primary_issuance: bool) {
+            self.owner.set(self.env().caller());
+            self.max_allowed_percentage.set(max_percentage);
+            self.allow_primary_issuance.set(primary_issuance);
+        }
+
+        /// This function is used to verify transfers initiated by the
+        /// runtime assets
+        ///
+        /// # Arguments
+        /// * `from` - Identity Id of the sender.
+        /// * `to` - Identity Id of the receiver.
+        /// * `value` - Asset amount need to transfer to the receiver.
+        /// * `balance_from` - Balance of sender at the time of transaction.
+        /// * `balance_to` - Balance of receiver at the time of transaction.
+        /// * `total_supply` - Total supply of the asset
+        #[ink(message)]
+        fn verify_transfer(
+            &self,
+            // from: IdentityId,
+            // to: IdentityId,
+            value: Balance,
+            balance_from: Balance,
+            balance_to: Balance,
+            total_supply: Balance
+        ) -> RestrictionResult {
+            if /*from == None &&*/ *self.allow_primary_issuance.get() {
+                return RestrictionResult::Valid;
+            }
+            // if *self.exemption_list.get(to) {
+            //     return RestrictionResult::Valid;
+            // }
+            if ((balance_to + value) * 10u128.pow(6)) / total_supply > *self.max_allowed_percentage.get() {
+                return RestrictionResult::Invalid;
+            } else {
+                return RestrictionResult::Valid;
+            }
```

Comment:
It can be simplified to something like 
```
if (!issuance && amount > allowed) {
    return RestrictionResult::Invalid;
}
return RestrictionResult::Valid;
```